Kernel/FileSystem/RAMFS/Inode.cpp

```diff
@@ -122,6 +122,7 @@ ErrorOr<void> RAMFSInode::ensure_allocated_blocks(size_t offset, size_t io_size)
         }
     }
     clean_allocated_blocks_on_failure.disarm();
+    m_metadata.block_count = m_blocks.size();
```

Comment:
Does it fix the issue actually? I don't see where we set the block size for `RAMFS` so it will essentially multiply the block count by 0, isn't it correct?

Reply:
Block size is assumed to be 512/1024/"whatever the user puts in" by `du`. I don't think POSIX has a facility to retrieve the filesystem block size?

Reply:
Addendum: Ah, yeah, there is `st_blksize`. If we have that implemented then we should make sure that it's set.

Reply:
st_blksize is being set by two classes down the line, in BlockBasedFileSystem.h (precisely [here](https://github.com/SerenityOS/serenity/blob/808c98511b1e45cf1ccc05c0302270119edb25a6/Kernel/FileSystem/BlockBasedFileSystem.h#L42). I checked, and it's always 512. Is this enough, or should we for some reason set it again in this Inode.cpp?

Reply:
`RAMFS` is not inheriting from `BlockBasedFileSystem`, so setting this there is probably not affecting us anyway.

Each data block is using a 128 KiB space, so 512 bytes is probably incorrect.
https://github.com/SerenityOS/serenity/blob/ad4bd36c59ea2720451f729dcb672c8bcfbd0ea0/Kernel/FileSystem/RAMFS/Inode.h#L69-L75

In #19867, I actually reduced this to 4 KiB. But 128 KiB is way too big to do anything useful with it.
Maybe change the block size to PAGE_SIZE:
```
constexpr static size_t data_block_size = PAGE_SIZE;
```
and then change st_blksize by setting it to that size as well.

Reply:
oops, mb. I'm not sure where it gets the 512 from then (I did verify that it does)

will push a fix later